src/main/kotlin/com/zopa/ktor/opentracing/OpenTracingServer.kt

```diff
@@ -82,6 +84,20 @@ class OpenTracingServer {
                 }
             }
 
+            pipeline.environment.monitor.subscribe(Routing.RoutingCallFinished) {call ->
+                if (config.filters.any { it(call) }) return@subscribe
+                val span = GlobalTracer.get().activeSpan()
+
+                val routePath = call.route.parent.toString()
+                val routeMethod = (call.route.selector as HttpMethodRouteSelector).method.value
```

Comment:
we need to handle the unsafe cast here , maybe try to parse it from the span name that was set in the first intercept and otherwise just leave it as is by default

Reply:
Lol juuust changed this